aria/orchestrator/workflows/builtin/utils.py

```diff
@@ -14,33 +14,38 @@
 # limitations under the License.
 
 from ..api.task import OperationTask
+from .. import exceptions
 
 
-def create_node_task(operation_name, node):
+def create_node_task(interface_name, operation_name, node):
     """
     Returns a new operation task if the operation exists in the node, otherwise returns None.
     """
 
-    if _has_operation(node.interfaces, operation_name):
-        return OperationTask.node(instance=node,
-                                  name=operation_name)
-    return None
+    try:
+        return OperationTask.for_node(node=node,
+                                      interface_name=interface_name,
+                                      operation_name=operation_name)
+    except exceptions.TaskException:
```

Comment:
OK. This means that I will need to create a new exception sub-class here, specifically for task creation.